py/picca/io.py

```diff
@@ -178,19 +200,12 @@ def read_data(in_dir,drq,mode,zmin = 2.1,zmax = 3.5,nspec=None,log=None,keep_bal
             t0 = time.time()
             pix_data = read_from_spec(in_dir,thid[w], ra[w], dec[w], zqso[w], plate[w], mjd[w], fid[w], order, mode=mode,log=log)
             read_time=time.time()-t0
-        elif mode =="spcframe":
-            t0 = time.time()
-            pix_data = read_from_spcframe(in_dir,thid[w], ra[w], dec[w], zqso[w], plate[w], mjd[w], fid[w], order, mode=mode, log=log)
-            read_time=time.time()-t0
         if not pix_data is None:
             sys.stderr.write("{} read from pix {}, {} {} in {} secs per spectrum\n".format(len(pix_data),pix,i,len(upix),read_time/(len(pix_data)+1e-3)))
         if not pix_data is None and len(pix_data)>0:
             data[pix] = pix_data
             ndata += len(pix_data)
 
-        if not nspec is None:
-            if ndata > nspec:break
```

Comment:
why do you remove this?

Reply:
this is now done between l. 116 and l.124:

    if nspec != None:
        ra = ra[:nspec]
        dec = dec[:nspec]
        zqso = zqso[:nspec]
        thid = thid[:nspec]
        plate = plate[:nspec]
        mjd = mjd[:nspec]
        fid = fid[:nspec]